src/test_gpio.sh

```diff
@@ -0,0 +1,214 @@
+#!/bin/sh
+#
+# Copyright (c) 2016, Imagination Technologies Limited and/or its affiliated group companies
+# and/or licensors
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification, are permitted
+# provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+#    and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+#    conditions and the following disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+#    endorse or promote products derived from this software without specific prior written
+#    permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+# FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+# WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# This test implements 16.3, 16.4, 16.5 and 16.6 sections of GATE Test procedure specifications.
+
+
+. /usr/lib/board_test_common.sh
+
+NR_ERROR=0
+
+parse_command_line $@
+
+redirect_output_and_error $LOG_LEVEL
+
+check_mfio()
+{
+	local MFIO_NO=$1
+	local EXPECTED=$2
+
+	local VALUE=$(./test_get_pin.sh $MFIO_NO)
+	if [ $VALUE -ne $EXPECTED ]; then
+		LOG_ERROR "mfio $MFIO_NO expected value is $EXPECTED but got $VALUE"
+		NR_ERROR=$((NR_ERROR + 1))
+	fi
+}
+
+LOG_INFO "16.3.1 MIKRO1-2_Loop23-27Lo"
```

Comment:
This test is production test supposed to be run on a special jig. don't think it will add any value for a generic user who will not have this setup.